File: api/pkg/diplomacy/order.go
package diplomacy

import "fmt"

// OrderType represents the type of order a unit can be given.
type OrderType int

const (
	OrderHold    OrderType = iota // Unit holds position
	OrderMove                     // Unit moves to adjacent province
	OrderSupport                  // Unit supports another unit's hold or move
	OrderConvoy                   // Fleet convoys army across sea
)

func (o OrderType) String() string {
	switch o {
	case OrderHold:
		return "hold"
	case OrderMove:
		return "move"
	case OrderSupport:
		return "support"
	case OrderConvoy:
		return "convoy"
	default:
		return "unknown"
	}
}

// Order represents a single order issued to a unit.
type Order struct {
	// Unit being ordered
	UnitType UnitType
	Power    Power
	Location string
	Coast    Coast // Coast of the unit being ordered (for fleets on split coasts)

	// Order details
	Type OrderType

	// Target province (for move, support, convoy)
	Target      string
	TargetCoast Coast // Coast of target (for fleet moves to split-coast provinces)

	// Aux fields for support and convoy:
	// For support: the province the supported unit is in
	// For convoy: the province the convoyed army is in
	AuxLoc string
	// For support: the destination the supported unit is moving to (empty if support-hold)
	// For convoy: the destination the convoyed army is moving to
	AuxTarget string
	// For support: the type of the supported unit
	AuxUnitType UnitType
}

// OrderResult describes the outcome of adjudicating an order.
type OrderResult int

const (
	ResultSucceeded OrderResult = iota // Order carried out
	ResultFailed                       // Move bounced or support failed to enable
	ResultDislodged                    // Unit was dislodged
	ResultBounced                      // Move bounced
	ResultCut                          // Support was cut
	ResultVoid                         // Order was invalid, treated as hold
)

func (r OrderResult) String() string {
	switch r {
	case ResultSucceeded:
		return "succeeded"
	case ResultFailed:
		return "failed"
	case ResultDislodged:
		return "dislodged"
	case ResultBounced:
		return "bounced"
	case ResultCut:
		return "cut"
	case ResultVoid:
		return "void"
	default:
		return "unknown"
	}
}

// ResolvedOrder pairs an order with its adjudication result.
type ResolvedOrder struct {
	Order  Order
	Result OrderResult
}

// Describe returns a human-readable description of the order.
func (o *Order) Describe() string {
	unitStr := "A"
	if o.UnitType == Fleet {
		unitStr = "F"
	}
	loc := o.Location
	if o.Coast != NoCoast {
		loc += "/" + string(o.Coast)
	}

	switch o.Type {
	case OrderHold:
		return fmt.Sprintf("%s %s Hold", unitStr, loc)
	case OrderMove:
		target := o.Target
		if o.TargetCoast != NoCoast {
			target += "/" + string(o.TargetCoast)
		}
		return fmt.Sprintf("%s %s -> %s", unitStr, loc, target)
	case OrderSupport:
		auxUnit := "A"
		if o.AuxUnitType == Fleet {
			auxUnit = "F"
		}
		if o.AuxTarget == "" {
			return fmt.Sprintf("%s %s S %s %s Hold", unitStr, loc, auxUnit, o.AuxLoc)
		}
		return fmt.Sprintf("%s %s S %s %s -> %s", unitStr, loc, auxUnit, o.AuxLoc, o.AuxTarget)
	case OrderConvoy:
		return fmt.Sprintf("%s %s C A %s -> %s", unitStr, loc, o.AuxLoc, o.AuxTarget)
	default:
		return fmt.Sprintf("%s %s ???", unitStr, loc)
	}
}

File: ui/lib/core/models/order.dart
/// Order returned from the API (snake_case JSON).
class Order {
  final String id;
  final String phaseId;
  final String power;
  final String unitType;
  final String location;
  final String orderType;
  final String? target;
  final String? auxLoc;
  final String? auxTarget;
  final String? auxUnitType;
  final String? result;
  final DateTime createdAt;

  const Order({
    required this.id,
    required this.phaseId,
    required this.power,
    required this.unitType,
    required this.location,
    required this.orderType,
    this.target,
    this.auxLoc,
    this.auxTarget,
    this.auxUnitType,
    this.result,
    required this.createdAt,
  });

  factory Order.fromJson(Map<String, dynamic> json) {
    return Order(
      id: json['id'] as String,
      phaseId: json['phase_id'] as String,
      power: json['power'] as String,
      unitType: json['unit_type'] as String,
      location: json['location'] as String,
      orderType: json['order_type'] as String,
      target: json['target'] as String?,
      auxLoc: json['aux_loc'] as String?,
      auxTarget: json['aux_target'] as String?,
      auxUnitType: json['aux_unit_type'] as String?,
      result: json['result'] as String?,
      createdAt: DateTime.parse(json['created_at'] as String),
    );
  }
}

/// Order input for submission to backend.
class OrderInput {
  final String unitType;
  final String location;
  final String? coast;
  final String orderType;
  final String? target;
  final String? targetCoast;
  final String? auxLoc;
  final String? auxTarget;
  final String? auxUnitType;

  const OrderInput({
    required this.unitType,
    required this.location,
    this.coast,
    required this.orderType,
    this.target,
    this.targetCoast,
    this.auxLoc,
    this.auxTarget,
    this.auxUnitType,
  });

  Map<String, dynamic> toJson() {
    final json = <String, dynamic>{
      'unit_type': unitType,
      'location': location,
      'order_type': orderType,
    };
    if (coast != null) json['coast'] = coast;
    if (target != null) json['target'] = target;
    if (targetCoast != null) json['target_coast'] = targetCoast;
    if (auxLoc != null) json['aux_loc'] = auxLoc;
    if (auxTarget != null) json['aux_target'] = auxTarget;
    if (auxUnitType != null) json['aux_unit_type'] = auxUnitType;
    return json;
  }
}
